pkg/virt-launcher/virtwrap/converter/converter.go

```diff
@@ -1621,7 +1633,10 @@ func Convert_v1_VirtualMachineInstance_To_api_Domain(vmi *v1.VirtualMachineInsta
 
 		hpStatus, hpOk := c.HotplugVolumes[disk.Name]
 		// if len(c.PermanentVolumes) == 0, it means the vmi is not ready yet, add all disks
-		if _, ok := c.PermanentVolumes[disk.Name]; ok || len(c.PermanentVolumes) == 0 || (hpOk && (hpStatus.Phase == v1.HotplugVolumeMounted || hpStatus.Phase == v1.VolumeReady)) {
+		if _, ok := c.PermanentVolumes[disk.Name]; ok ||
+			len(c.PermanentVolumes) == 0 ||
+			volume == nil ||
+			(hpOk && (hpStatus.Phase == v1.HotplugVolumeMounted || hpStatus.Phase == v1.VolumeReady)) {
```

Comment:
I got lost here with all the conditions. Can you please use a variable and name properly to the check?